### ISSUE TITLE
JSParser fails on defineProperty getters

### ISSUE THREAD

**Issue by JasonTheProgrammer** [trullock/NUglify#120]
Testing on master.

The following:
```
var result = Uglify.Js("var o = {}; Object.defineProperty(o, 'a', { get() { return 'foobar world'; } });");
```

Throws a nullref exception when attempting to construct the `GetterSetter`.
https://github.com/trullock/NUglify/blob/master/src/NUglify/JavaScript/JSParser.cs#L4799

```
field = new GetterSetter(funcExpr.Binding.Name, isGet, funcExpr.Binding.Context.Clone());
```

The getter has no function name (as is valid syntax, per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty), so `ParseFunction` gives a result with no `Binding` set.

I think the problem ultimately lies in ParseFunction.
https://github.com/trullock/NUglify/blob/master/src/NUglify/JavaScript/JSParser.cs#L2842

Here it fails to set a binding for the function. The comments (wrongly) state "if this isn't a function expression, then we need to throw an error because function DECLARATIONS always need a valid identifier name".

A simple kludge is to insert the following on line 2845:

```
if (m_currentToken.Is(JSToken.LeftParenthesis))
{
    name = new BindingIdentifier(CurrentPositionContext)
    {
        Name = ""
    };
}
```

But I'm conscious that this is probably only valid in the the context of Object.defineProperty, so there should maybe be something further up the stack to identify that context and pass it through to ParseFunction?

**Comment by trullock** [comment]
Sorry I missed this, my notification settings were screwed.

Thanks for the heads up, definitely a bug and your kludge really helped my identify the issue quickly.

However, its not quite the right fix as you suspect. 

PR to follow with a more complete solution, would appreciate some QA!

**Comment by trullock** [comment]
Shipped in v1.8.1
